front/src/modules/conflict/components/ConflictCard.tsx

```diff
@@ -19,16 +24,69 @@ const ConflictCard = ({
   const start_time = new Date(conflict.start_time).toLocaleTimeString(dateTimeLocale);
   const end_time = new Date(conflict.end_time).toLocaleTimeString(dateTimeLocale);
   const start_date = new Date(conflict.start_time).toLocaleDateString(dateTimeLocale);
+  const dispatch = useAppDispatch();
+  const [visibleTrains, setVisibleTrains] = useState<number[]>([
+    ...Array(conflict.trainNames.length).keys(),
+  ]);
+  const [showOthers, setShowOthers] = useState(false);
+  const [isOthersTooltipOpen, setIsOthersTooltipOpen] = useState(false);
+  const trainsContainerRef = useRef<HTMLDivElement>(null);
+  const othersTagRef = useRef<HTMLDivElement>(null);
 
   const subCategories = useSubCategoryContext();
 
+  const trainIds = useMemo(() => {
+    const scheduleIds = conflict.train_schedule_ids.map(
+      (id) => `trainschedule_${id}` as TrainScheduleId
```

Comment:
Same for occurrences below
```suggestion
      (id) => formatEditoastIdToTrainScheduleId(id)
```

Reply:
Done